urth/dashboard/converter/static/urth/dashboard.js

```diff
@@ -17,53 +25,46 @@ define(['jquery', 'Thebe', 'urth-common/error-log'], function($, Thebe, ErrorLog
-
-        // show a busy indicator when communicating with kernel
-        var debounced;
-        thebe.events.on('kernel_busy.Kernel kernel_idle.Kernel', function(event) {
-            clearTimeout(debounced);
-            debounced = setTimeout(function() {
-                var isBusy = event.type === 'kernel_busy';
-                $('.busy-indicator')
-                    .toggleClass('show', isBusy)
-                    // Prevent progress animation when hidden by removing 'active' class.
-                    .find('.progress-bar')
-                        .toggleClass('active', isBusy);
-            }, 500);
-        });
+    function initKernel(args) {
+        var kernelReady = $.Deferred();
+
+        // TODO
+        // 1. Figure out how to load `display_data` comm messages (includes source for matplotlib, styling, etc)
+        // 2. Create this global, so extensions like matplotlib can register a comm callback:
+        //      IPython.notebook.kernel.comm_manager.register_target('matplotlib', mpl.mpl_figure_comm)
+
+        var options = {
+            baseUrl: 'http://192.168.99.100:9500',
+            wsUrl: 'ws://192.168.99.100:9500',
+            name: 'python3'
+        };
+        Services.startNewKernel(options).then(function(_kernel) {
+            kernel = _kernel;
+
+            // show a busy indicator when communicating with kernel
+            var debounced;
+            kernel.statusChanged.connect(function(_kernel, status) {
+                clearTimeout(debounced);
+                debounced = setTimeout(function() {
+                    var isBusy = status === Services.KernelStatus.Busy;
+                    $('.busy-indicator')
+                        .toggleClass('show', isBusy)
+                        // Prevent progress animation when hidden by removing 'active' class.
+                        .find('.progress-bar')
+                            .toggleClass('active', isBusy);
+                }, 500);
+            });
+            kernel.commOpened.connect(function(_kernel, commMsg) {
+                var comm = kernel.connectToComm(commMsg.target_name, commMsg.comm_id);
+            });
 
-        // override this function since it messes with images (invoking jquery.resizable on them)
-        IPython.OutputArea.prototype._dblclick_to_reset_size = function() {};
+            // TODO IPython.notebook.events.emit('kernel_connected.Kernel', {data: kernel});
+            //          ==> IPyWidgets expects `kernel` to have `.comm_manager.new_comm`
 
-        // hook the error handler to kernel / session events
-        thebe.events.on('kernel_created.Kernel kernel_created.Session', function(e) {
-            ErrorLog.enable(window.IPython);
+            kernelReady.resolve();
         });
 
-        // finally, let's start the kernel (rather than waiting for it to be lazily loaded)
-        var kernel_ready = new $.Deferred();
-        if(Urth.tmpnb_mode) {
-            thebe.call_spawn(kernel_ready.resolve);
```

Comment:
Reminder: Need to implement spawn ahead of kernel request if we're sticking with tmpnb, or binder, or some other container provisioner.